plugins/org.polarsys.capella.cybersecurity.ui/src/org/polarsys/capella/cybersecurity/ui/contribution/migration/CybersecurityMigrationContribution.java

```diff
@@ -81,27 +85,30 @@ public boolean ignoreUnknownFeature(String prefix, String name, boolean isElemen
   @Override
   public void dispose(MigrationContext context) {
     savedValues.clear();
-    project = null;
-    isCyberViewpointActive = false;
+    config = null;
   }
 
   private void threatMigration(Threat threat) {
-    EnumerationPropertyLiteral literal = getLiteral(CybersecurityQueries.getThreatKindPropertyType(project),
-        savedValues.get(threat.hashCode() + ":threatKind"));
+    String searchValue = savedValues.get(threat.hashCode() + ":threatKind");
+    // threat kind "Theft and data alteration" was split in "Theft" and "Data alteration"
+    if (searchValue != null && searchValue.contains("THEFT")) {
+      searchValue = "THEFT";
+    }
 
+    EnumerationPropertyLiteral literal = getLiteral(config.getThreatKind(), searchValue);
     if (literal != null) {
       threat.setKind(literal);
     }
   }
 
   private void securityNeedsMigration(SecurityNeeds sn) {
-    EnumerationPropertyLiteral literalC = getLiteral(CybersecurityQueries.getConfidentialityPropertyType(project),
+    EnumerationPropertyLiteral literalC = getLiteral(config.getConfidentiality(),
         savedValues.get(sn.hashCode() + ":confidentiality"));
```

Comment:
When you use hashCode, can you use getId() instead? 
several objects can have the same hashCode

Reply:
Updated.